test/TestProjects/HeadAsBooleanTrue-LowLevel/Generated/HttpSuccessClient.cs

```diff
@@ -159,47 +159,42 @@ public virtual Response<bool> Head404(RequestContext context = null)
 
         internal HttpMessage CreateHead200Request(RequestContext context)
         {
-            var message = _pipeline.CreateMessage(context);
+            var message = _pipeline.CreateMessage(context, ResponseClassifier200To300400To500);
             var request = message.Request;
             request.Method = RequestMethod.Head;
             var uri = new RawRequestUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/http/success/200", false);
             request.Uri = uri;
-            message.ResponseClassifier = ResponseClassifier200To300400To500.Instance;
             return message;
         }
 
         internal HttpMessage CreateHead204Request(RequestContext context)
         {
-            var message = _pipeline.CreateMessage(context);
+            var message = _pipeline.CreateMessage(context, ResponseClassifier200To300400To500);
             var request = message.Request;
             request.Method = RequestMethod.Head;
             var uri = new RawRequestUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/http/success/204", false);
             request.Uri = uri;
-            message.ResponseClassifier = ResponseClassifier200To300400To500.Instance;
             return message;
         }
 
         internal HttpMessage CreateHead404Request(RequestContext context)
         {
-            var message = _pipeline.CreateMessage(context);
+            var message = _pipeline.CreateMessage(context, ResponseClassifier200To300400To500);
             var request = message.Request;
             request.Method = RequestMethod.Head;
             var uri = new RawRequestUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/http/success/404", false);
             request.Uri = uri;
-            message.ResponseClassifier = ResponseClassifier200To300400To500.Instance;
             return message;
         }
 
-        private sealed class ResponseClassifier200To300400To500 : ResponseClassifier
+        private sealed class ResponseClassifier200To300400To500Override : ResponseClassifier
         {
-            private static ResponseClassifier _instance;
-            public static ResponseClassifier Instance => _instance ??= new ResponseClassifier200To300400To500();
             public override bool IsErrorResponse(HttpMessage message)
             {
                 return message.Response.Status switch
```

Comment:
[Here](https://github.com/Azure/autorest.csharp/blob/ebc6eede5294c2b0300bf86b28e852c5fa3a4f03/test/TestProjects/HeadAsBooleanTrue-LowLevel/Generated/HttpSuccessClient.cs#L196-L210) is the generated code for the `ResponseClassifier200To300400To500` bases on https://github.com/Azure/autorest.csharp/issues/2006#issuecomment-1055887869